spyder/plugins/variableexplorer/widgets/namespacebrowser.py

```diff
@@ -392,3 +415,20 @@ def save_data(self, filename=None):
                     '<br><br>Error message:<br>') + error_message
             QMessageBox.critical(self, _("Save data"), save_data_message)
         self.save_button.setEnabled(self.filename is not None)
+
+
+class NamespacesBrowserFinder(FinderLineEdit):
+    """Textbox for filtering listed variables in the table."""
+
+    def keyPressEvent(self, event):

```

Comment:
Please add here the possibility to hide this widget when users press the `Esc` key.